src/main/webapp/components/LoadingWrapper.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import { div, img } from 'react-hyperscript-helpers';
+import { context } from "../util/UrlConstants";
+
+function LoadingWrapper(WrappedComponent, overlaySpinner = '') {
+  const baseStyle = {
+    'left': '0',
+    'width': '100%',
+    'height': '100%',
+    'background': 'rgba(255, 255, 255, 0.5)',
+  };
+
+  const containerStyle = {
+    'position': 'absolute',
+    'top': '51px',
+    'zIndex': '1029'
+  };
+
+  const containerStyleFixed = {
+    'position': 'fixed',
+    'top': '0',
+    'zIndex': '1041'
+  };
+
+  const containerStyleFixedTop = {
+    'position': 'fixed',
+    'top': '51px',
+    'zIndex': '1041'
+  };
+
+  const spinnerStyle = {
+    'position': 'absolute',
+    'top': '30%',
+    'left': '50%',
+    'margin': '-30px 0 0 -30px',
+    'zIndex': '9999'
+  };
+
+
+
+  return function WithLoadingComponent({...props}) {
+
+    const [loading, setLoading] = useState(false);
+
+    const showSpinner = () => {
+      setLoading(true);
+    };
+
+    const hideSpinner = () => {
+      setLoading(false);
+    };
+
+    const getStyle = () => {
+      let style = {};
+      if (overlaySpinner === 'fixed') {
+        style = {...baseStyle, ...containerStyleFixed }
+      } else if (overlaySpinner === 'fixedTop') {
```

Comment:
Would be nice if these were constants defined somewhere. I can easily see bugs cropping up where this series of if's is expanded and devs mistype something. Also, might be better structured as a switch block.

Reply:
It's true, eitherway we discussed styles with @jaime-bernales-broad and @wloforteBroad and we decided to remove one of these styles for a better user experience. The result is true or false condition now, so these constants are not needed anymore.